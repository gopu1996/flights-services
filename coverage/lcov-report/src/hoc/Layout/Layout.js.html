<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/hoc/Layout/Layout.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/hoc/Layout</a> Layout.js
    </h1>
    <div class='clearfix'>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Aux from "../AuxHoc/Auxs";
import Toolbar from "../../components/Navigation/Toolbar/Toolbar";
import React, { Component } from "react";
import { connect } from "react-redux";
&nbsp;
class Layout extends Component {
<span class="fstat-no" title="function not covered" >  re</span>nder() {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;Aux&gt;
        &lt;Toolbar
          isAuth={this.props.isAuthenticated}
          isUserName={this.props.firstName}
          isUserType={this.props.userType}
        /&gt;
        &lt;main&gt;{this.props.children}&lt;/main&gt;
      &lt;/Aux&gt;
    );
  }
}
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    isAuthenticated: state.auth.token !== null,
    firstName: state.auth.firstName,
    userType: state.auth.userType,
    expiredIn: state.auth.expiredIn,
  };
};
&nbsp;
export default connect(mapStateToProps)(Layout);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 28 2020 15:53:54 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
