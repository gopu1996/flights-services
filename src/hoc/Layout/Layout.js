import Aux from "../AuxHoc/Auxs";
import Toolbar from "../../components/Navigation/Toolbar/Toolbar";
import React, { Component } from "react";
import { connect } from "react-redux";

class Layout extends Component {
  render() {
    return (
      <Aux>
        <Toolbar
          isAuth={this.props.isAuthenticated}
          isUserName={this.props.firstName}
          isUserType={this.props.userType}
        />
        <main>{this.props.children}</main>
      </Aux>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    isAuthenticated: state.auth.token !== null,
    firstName: state.auth.firstName,
    userType: state.auth.userType,
    expiredIn: state.auth.expiredIn,
  };
};

export default connect(mapStateToProps)(Layout);
