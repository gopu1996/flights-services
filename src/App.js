import React from "react";
import Layout from "./hoc/Layout/Layout";
import { Route, Switch } from "react-router-dom";
import Login from "./containers/Auth/Login/Login";
import Logout from "./containers/Auth/logout/Logout";
import LazyLoading from "./hoc/asyncComponent/LazyLoading";

const FlightDetailsContainer = LazyLoading(() => {
  return import("./containers/flightDetails/FlightDetailsCointainer");
});

const FlightListCointainer = LazyLoading(() => {
  return import("./containers/flightlists/FlightListsCointainer");
});

function App() {
  return (
    <div className="App">
      <Layout>
        <Switch>
          <Route path="/" exact component={Login}  />
          <Route path="/logout" exact component={Logout} />\
          <Route path="/flightList" exact component={FlightListCointainer} />
          <Route
            path="/flightdetails/:passFlightId"
            component={FlightDetailsContainer}
          />
        </Switch>
      </Layout>
    </div>
  );
}

export default App;
