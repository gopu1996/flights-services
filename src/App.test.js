import Adapter from "enzyme-adapter-react-16";
import React from 'react';
import { shallow, configure } from 'enzyme';
import App from './App';
import Logout from "./containers/Auth/logout/Logout";
import FlightListCointainer from "./containers/flightlists/FlightListsCointainer";
import { Route } from 'react-router';
import Layout from './hoc/Layout/Layout'
import * as renderer from "react-test-renderer";

configure({ adapter: new Adapter() });
let component;
describe('Validating Router in App', () => {
  beforeEach(() => {
    component = shallow(<App/>);
});
    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });
  

    it('should show flightList component for /flightList router', () => {
        const wrapper = shallow(<App />);
        const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
          const routeProps = route.props();
          pathMap[routeProps.path] = routeProps.path;
          return pathMap;

        }, {});  
        expect(pathMap['/flightList']).toBe('/flightList');
      });

it('should show Login component for / router', () => {
    const wrapper = shallow(<App />);
    const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.path;
      return pathMap;
    }, {});  
    expect(pathMap['/']).toBe('/');
  });

  it('should show logout component for /logout router', () => {
    const wrapper = shallow(<App />);
    const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});  
    expect(pathMap['/logout']).toBe(Logout);
  });

  it('should show Flightdetails component for /flightdetails/:passFlightId router' , () => {
    const wrapper = shallow(<App />);
    const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.path;
      return pathMap;
    }, {});  
    expect(pathMap['/flightdetails/:passFlightId']).toBe('/flightdetails/:passFlightId');
  });


})