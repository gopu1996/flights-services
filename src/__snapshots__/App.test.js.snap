// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validating Router in App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown>
        <Switch>
                <Route
                        component={
                                Object {
                                        "$$typeof": Symbol(react.memo),
                                        "WrappedComponent": [Function],
                                        "compare": null,
                                        "displayName": "Connect(Login)",
                                        "type": [Function],
                                        Symbol(Symbol.hasInstance): [Function],
                                      }
                        }
                        exact={true}
                        path="/"
                />
                <Route
                        component={
                                Object {
                                        "$$typeof": Symbol(react.memo),
                                        "WrappedComponent": [Function],
                                        "compare": null,
                                        "displayName": "Connect(Logout)",
                                        "type": [Function],
                                        Symbol(Symbol.hasInstance): [Function],
                                      }
                        }
                        exact={true}
                        path="/logout"
                />
                \\
                <Route
                        component={[Function]}
                        exact={true}
                        path="/flightList"
                />
                <Route
                        component={[Function]}
                        path="/flightdetails/:passFlightId"
                />
        </Switch>
</Unknown>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Switch>
          <Route
                    component={
                              Object {
                                        "$$typeof": Symbol(react.memo),
                                        "WrappedComponent": [Function],
                                        "compare": null,
                                        "displayName": "Connect(Login)",
                                        "type": [Function],
                                        Symbol(Symbol.hasInstance): [Function],
                                      }
                    }
                    exact={true}
                    path="/"
          />
          <Route
                    component={
                              Object {
                                        "$$typeof": Symbol(react.memo),
                                        "WrappedComponent": [Function],
                                        "compare": null,
                                        "displayName": "Connect(Logout)",
                                        "type": [Function],
                                        Symbol(Symbol.hasInstance): [Function],
                                      }
                    }
                    exact={true}
                    path="/logout"
          />
          \\
          <Route
                    component={[Function]}
                    exact={true}
                    path="/flightList"
          />
          <Route
                    component={[Function]}
                    path="/flightdetails/:passFlightId"
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={
                            Object {
                                          "$$typeof": Symbol(react.memo),
                                          "WrappedComponent": [Function],
                                          "compare": null,
                                          "displayName": "Connect(Login)",
                                          "type": [Function],
                                          Symbol(Symbol.hasInstance): [Function],
                                        }
              }
              exact={true}
              path="/"
/>,
            <Route
              component={
                            Object {
                                          "$$typeof": Symbol(react.memo),
                                          "WrappedComponent": [Function],
                                          "compare": null,
                                          "displayName": "Connect(Logout)",
                                          "type": [Function],
                                          Symbol(Symbol.hasInstance): [Function],
                                        }
              }
              exact={true}
              path="/logout"
/>,
            "\\\\",
            <Route
              component={[Function]}
              exact={true}
              path="/flightList"
/>,
            <Route
              component={[Function]}
              path="/flightdetails/:passFlightId"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Login)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Logout)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
              "exact": true,
              "path": "/logout",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "\\\\",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/flightList",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/flightdetails/:passFlightId",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Layout)",
        "type": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown>
          <Switch>
                    <Route
                              component={
                                        Object {
                                                  "$$typeof": Symbol(react.memo),
                                                  "WrappedComponent": [Function],
                                                  "compare": null,
                                                  "displayName": "Connect(Login)",
                                                  "type": [Function],
                                                  Symbol(Symbol.hasInstance): [Function],
                                                }
                              }
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={
                                        Object {
                                                  "$$typeof": Symbol(react.memo),
                                                  "WrappedComponent": [Function],
                                                  "compare": null,
                                                  "displayName": "Connect(Logout)",
                                                  "type": [Function],
                                                  Symbol(Symbol.hasInstance): [Function],
                                                }
                              }
                              exact={true}
                              path="/logout"
                    />
                    \\
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/flightList"
                    />
                    <Route
                              component={[Function]}
                              path="/flightdetails/:passFlightId"
                    />
          </Switch>
</Unknown>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Switch>
            <Route
                        component={
                                    Object {
                                                "$$typeof": Symbol(react.memo),
                                                "WrappedComponent": [Function],
                                                "compare": null,
                                                "displayName": "Connect(Login)",
                                                "type": [Function],
                                                Symbol(Symbol.hasInstance): [Function],
                                              }
                        }
                        exact={true}
                        path="/"
            />
            <Route
                        component={
                                    Object {
                                                "$$typeof": Symbol(react.memo),
                                                "WrappedComponent": [Function],
                                                "compare": null,
                                                "displayName": "Connect(Logout)",
                                                "type": [Function],
                                                Symbol(Symbol.hasInstance): [Function],
                                              }
                        }
                        exact={true}
                        path="/logout"
            />
            \\
            <Route
                        component={[Function]}
                        exact={true}
                        path="/flightList"
            />
            <Route
                        component={[Function]}
                        path="/flightdetails/:passFlightId"
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={
                                Object {
                                                "$$typeof": Symbol(react.memo),
                                                "WrappedComponent": [Function],
                                                "compare": null,
                                                "displayName": "Connect(Login)",
                                                "type": [Function],
                                                Symbol(Symbol.hasInstance): [Function],
                                              }
                }
                exact={true}
                path="/"
/>,
              <Route
                component={
                                Object {
                                                "$$typeof": Symbol(react.memo),
                                                "WrappedComponent": [Function],
                                                "compare": null,
                                                "displayName": "Connect(Logout)",
                                                "type": [Function],
                                                Symbol(Symbol.hasInstance): [Function],
                                              }
                }
                exact={true}
                path="/logout"
/>,
              "\\\\",
              <Route
                component={[Function]}
                exact={true}
                path="/flightList"
/>,
              <Route
                component={[Function]}
                path="/flightdetails/:passFlightId"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Login)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Logout)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "exact": true,
                "path": "/logout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "\\\\",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/flightList",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/flightdetails/:passFlightId",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Layout)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
